include/aws/common/aws_profile.h

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#ifndef AWS_COMMON_AWS_PROFILE_H
```

Comment:
Protest note: aws-c-common is our "base C library". It doesn't have any AWS-SDK specific stuff in it right now. There are **certain things** built on top of aws-c-common that have nothing to do with SDKs

This PR is more of an "aws-c-auth-but-without-io-deps" kind of thing, but making another repo is a big time investment...

Do you foresee us taking more stuff out of aws-c-auth in the near future? Should we bite the bullet now and make aws-c-authbase? I don't want to just keep pulling things 1 by 1 out of aws-c-auth because at each individual step it was the least friction way to do things, and there's already a precedent for doing so